omod/src/main/webapp/resources/js/view/editors.js

```diff
@@ -368,7 +368,106 @@ define(
 		editors.LocationSelect = editors.GenericModelSelect.extend({
 			modelType: openhmis.Location,
 			displayAttr: "name",
-			allowNull: true
+			allowNull: true,
+			reOrderedLocations: new Backbone.Collection(),
+			
+			// Overriding renderOptions function to support hierarchical locations display
+            renderOptions: function(options) {
+            	options.models = this.reorderLocationsToSupportAllDepthChildrenNextToTheirParents(options.models);//TODO not assigning
+            	editors.GenericModelSelect.prototype.renderOptions.call(this, options);
+            },
+		
+		   /**
+			* Re-orders the locations collection models collection in such a way as to get all chidren next to their parents and supports depth
+			*/
+			reorderLocationsToSupportAllDepthChildrenNextToTheirParents: function(locationModels) {
+				var cur = this;
+				
+				if(locationModels !== null && locationModels !== undefined && locationModels.length > 0) {
+					var rootLocations = new Backbone.Collection();
+					
+					for(i = 0; i < locationModels.length; i++) {//look through all locations and pick out the root locations
+						var location = locationModels[i];
+						
+						if(location !== null && location !== undefined) {
+							location.fetch({async:false, success: function(fetchedloc) {//loads the location model with all its properties
+								if(fetchedloc !== null && fetchedloc !== undefined && (fetchedloc.get("parentLocation") === null || fetchedloc.get("parentLocation") === undefined)) {//true means fecthedLoc is a root location
+									rootLocations.add(location);
+								}
+							}});
+						}
+					}
+					
+					if(rootLocations !== null && rootLocations !== undefined && rootLocations.length > 0) {
+						for(i = 0; i < rootLocations.models.length; i++) {//for each root location, walk through the location tree
+							var rootLoc = rootLocations.models[i];
+							
+							if(rootLoc !== null && rootLoc !== undefined) {
+								var locationModel = cur.recreateRightModelObject(rootLoc.get("display"), rootLoc.get("uuid"), rootLoc.get("links")[0].uri);
+								
+								cur.addLocationToReorderedLocations(locationModel);//add the root location first before it's inherent children
+								cur.walkThroughTheLocationTree(rootLoc);
```

Comment:
Right, i did that and that's how i determined the problem, but why it happens, i couldn't establish that, my thought was that may be the self invoking logic was not good enough and brought about out of scope issues! but not sure what exactly was wrong